### ISSUE TITLE
Can't connect after disconecting

### ISSUE THREAD

**Issue by Thaodan** [TelepathyQt/telepathy-morse#61]
After authenticating via entering the pin code from telegram everything is fine.
But when disconnecting afterwards and reconnecting it doesn't work until the secret cache is removed.
My System:
Name                     : telepathy-morse-git
Version                  : r198.e8c2ef9-2
Branch build from  : dev
Name                     : telegram-qt-git
Version                  : r1141.1e45787-1
Branch build from : master
Distro                     : Arch Linux

Log:

https://paste.kde.org/pamwbxutq

**Comment by Thaodan** [comment]
I seems that the connection manager caches the secret when there's no pin code entered.

**Comment by Kaffeine** [comment]
Yes, that's true. At December I had the same issue and a workaround for the old API would require a complex logic. The issue is ruled out in the new API, I hope to roll out the first Morse port before April.

**Comment by Thaodan** [comment]
So just delete the cache everytime I get a dc or is there a better workaround?

What do you mean by new API?

**Comment by Kaffeine** [comment]
Probably fixed by commit https://github.com/Kaffeine/telegram-qt/commit/4c5eb3016269cd7596ce390aa12e2dfb8e209135

**Comment by Kaffeine** [comment]
The new API is growing in the ['next'](https://github.com/Kaffeine/telegram-qt/tree/next) branch. I'll let you know when it will be capable to receive text messages. :-) :-(

**Comment by Thaodan** [comment]
After logging sucsessfull, the 2nd time just get stuck but still asks for a key at telegram and doesn't ask for this key at tp

**Comment by Thaodan** [comment]
Log for the time when this happens [tp_morse.log](https://github.com/TelepathyIM/telepathy-morse/files/1925811/tp_morse.log)

**Comment by Thaodan** [comment]
Always this error:
bool CTelegramConnection::processRpcError(CTelegramStream&, quint64, TLValue) "RPC Error 420: FLOOD_WAIT_74696 for message 6545897377532202508 AuthSendCode (dc 2|149.154.167.51:443)"
RPC Error can not be handled.

**Comment by Thaodan** [comment]
I think the issue is that this is not handled:
https://core.telegram.org/api/errors#420-flood

**Comment by Thaodan** [comment]
[tp_morse_after_restart.log](https://github.com/TelepathyIM/telepathy-morse/files/1925827/tp_morse_after_restart.log)

**Comment by Thaodan** [comment]
Works now.
Except that after sometime no messsages come in.
